hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/ratis/OzoneManagerRatisServer.java

```diff
@@ -137,6 +141,59 @@ public OMResponse submitRequest(OMRequest omRequest) throws ServiceException {
     return processReply(omRequest, raftClientReply);
   }
 
+  /**
+   * Add a new OM to the Ratis ring.
+   * @return true if the new OM is successfully added to the ratis ring,
+   * false otherwise.
+   */
+  public boolean bootstrapNewOMs(List<OMNodeDetails> newOMNodes) {
+    StringBuilder newOMNodeIdsBuilder = new StringBuilder();
+    newOMNodes.stream().forEach(newOMNode ->
+        newOMNodeIdsBuilder.append(", ").append(newOMNode.getOMNodeId()));
+    String newOMNodeIds = newOMNodeIdsBuilder.toString().substring(2);
+    LOG.info("Bootstrapping new OM(s): {}", newOMNodeIds);
+
+    List<RaftPeer> newRaftPeerList = new ArrayList<>();
+    newRaftPeerList.addAll(raftPeers);
+
+    for (OMNodeDetails newOMNode : newOMNodes) {
+      String newOMNodeId = newOMNode.getOMNodeId();
+      RaftPeerId newOMRaftPeerId = RaftPeerId.valueOf(newOMNodeId);
+      InetSocketAddress newOMRatisAddr = new InetSocketAddress(
+          newOMNode.getHostName(), newOMNode.getRatisPort());
+
+      newRaftPeerList.add(RaftPeer.newBuilder()
+          .setId(newOMRaftPeerId)
+          .setAddress(newOMRatisAddr)
+          .build());
+    }
+
+    if (LOG.isDebugEnabled()) {
```

Comment:
nit: old log4j habit, we should not need this check as parameters are evaluated only if debug is enabled with this format. same is true for line 384

Reply:
Done